test/zeroD/test_zeroD.cpp

```diff
@@ -50,6 +52,185 @@ TEST(ZeroDim, test_individual_reactor_initialization)
+    }
+}
+
+TEST(MoleReactorTestSet, test_preconditioned_ideal_const_pressure)
+{
+    // Reactor 1
+    auto sol1 = newSolution("h2o2.yaml");
+    sol1->thermo()->setState_TP(1100.0, 10 * OneAtm);
+    sol1->thermo()->setEquivalenceRatio(1, "H2", "O2:0.21, N2:0.78, AR:0.09");
+    IdealGasConstPressureMoleReactor r1;
+    r1.insert(sol1);
+    // Reactor 2
+    auto sol2 = newSolution("h2o2.yaml");
+    sol2->thermo()->setState_TP(1100.0, 10 * OneAtm);
+    sol2->thermo()->setEquivalenceRatio(1, "H2", "O2:0.21, N2:0.78, AR:0.09");
+    IdealGasConstPressureMoleReactor r2;
+    r2.insert(sol2);
+    // Network 1
+    ReactorNet net1;
+    net1.addReactor(r1);
+    AdaptivePreconditioner precon;
+    net1.setProblemType(GMRES);
+    net1.setPreconditioner(precon);
+    // Network 2
+    ReactorNet net2;
+    net2.addReactor(r2);
+    net2.setProblemType(GMRES);
+    net2.initialize();
+    // Advancing
+    net1.advance(1.0);
+    net2.advance(1.0);
+    // Comparison
+    vector_fp state1(r1.neq(), 0.0);
+    vector_fp state2(r2.neq(), 0.0);
+    for (size_t i = 0; i < r1.neq(); i++)
+    {
+        EXPECT_NEAR(state1[i], state2[i], net1.atol());
+    }
+}
+
+TEST(MoleReactorTestSet, test_mole_reactor_get_state)
+{
+    // Setting up solution object and thermo/kinetics pointers
+    auto sol = newSolution("h2o2.yaml");
+    sol->thermo()->setState_TPY(1000.0, OneAtm, "H2:0.5, O2:0.5");
+    IdealGasConstPressureMoleReactor reactor;
+    reactor.insert(sol);
+    reactor.setInitialVolume(0.5);
+    reactor.setEnergy(false);
+    reactor.initialize();
+    vector_fp state(reactor.neq());
+    vector_fp updatedState(reactor.neq());
+    // test get state
+    const ThermoPhase& thermo = reactor.contents();
+    const vector_fp& imw = thermo.inverseMolecularWeights();
+    // prescribed state
+    double mass = reactor.volume() * thermo.density();
+    size_t H2I = reactor.componentIndex("H2")-1;
+    size_t O2I = reactor.componentIndex("O2")-1;
+    double O2_Moles = imw[O2I] * 0.5 * mass;
+    double  H2_Moles = imw[H2I] * 0.5 * mass;
+    // test getState
+    reactor.getState(state.data());
+    EXPECT_DOUBLE_EQ(state[reactor.componentIndex("H2")], H2_Moles);
+    EXPECT_DOUBLE_EQ(state[reactor.componentIndex("O2")], O2_Moles);
+    // Test updateState
+    reactor.updateState(state.data());
+    reactor.getState(updatedState.data());
+    EXPECT_DOUBLE_EQ(reactor.volume(), 0.5);
+    EXPECT_DOUBLE_EQ(reactor.pressure(), OneAtm);
+    for (size_t i = 0; i < reactor.neq(); i++)
+    {
+       EXPECT_DOUBLE_EQ(state[i], updatedState[i]);
+    }
+}
+
+TEST(MoleReactorTestSet, test_reinitialize_preconditioned_mole_reactors)
+{
+    // Solution object for reactors
+    auto sol1 = newSolution("h2o2.yaml");
+    sol1->thermo()->setState_TPX(300.0, OneAtm, "O2:1.0");
+    // Reactors
+    IdealGasMoleReactor r1;
+    IdealGasMoleReactor r2;
+    // Wall between the two
+    Wall wall;
+    wall.setArea(1.0);
+    wall.setHeatTransferCoeff(200);
```

Comment:
This wall never gets installed, so it has no effect on the system. As is, all this test does is check that nothing happens in these two reactors.